packages/rspack/src/compilation.ts

```diff
@@ -615,6 +616,71 @@ export class Compilation {
 		});
 	}
 
+	get chunks() {
+		var stats = this.getStats().toJson({
+			all: false,
+			chunks: true,
+			chunkModules: true,
+			reasons: true
+		});
+		const chunks = stats.chunks?.map(chunk => {
+			return {
+				...chunk,
+				name: chunk.names.length > 0 ? chunk.names[0] : "",
+				modules: this.__internal__getAssociatedModules(chunk)
+			};
+		});
```

Comment:
Now the complexity down to `O(mn)` , which is `chunk.length * modules count of each chunk`